client/packages/system/src/Report/ListView/ListView.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useState } from 'react';
+import {
+  TableProvider,
+  DataTable,
+  useColumns,
+  createTableStore,
+  NothingHere,
+  useTranslation,
+  useUrlQueryParams,
+  ReportContext,
+} from '@openmsupply-client/common';
+import { JsonData } from '@openmsupply-client/programs';
+import { useReport, ReportRowFragment } from '../api';
+import { Toolbar } from './Toolbar';
+import { ReportArgumentsModal } from '../components/ReportArgumentsModal';
+
+const ReportListComponent = ({ context }: { context: ReportContext }) => {
+  const {
+    filter,
+    updateSortQuery,
+    updatePaginationQuery,
+    queryParams: { sortBy, page, first, offset, filterBy },
+  } = useUrlQueryParams({
+    initialSort: { key: 'name', dir: 'asc' },
+    filterKey: 'name',
+  });
+  const queryParams = { filterBy, offset, sortBy };
+  const { data, isError, isLoading } = useReport.document.list({
+    context,
+    queryParams,
+  });
+  const pagination = { page, first, offset };
+  const t = useTranslation('common');
+  const [reportWithArgs, setReportWithArgs] = useState<
+    ReportRowFragment | undefined
+  >();
+
+  const columns = useColumns<ReportRowFragment>(
+    [
+      'name',
+      {
+        accessor: ({ rowData }) => rowData.context,
+        key: 'context',
+        label: 'label.context',
+        sortable: false,
+      },
+    ],
+    {
+      sortBy,
+      onChangeSortBy: updateSortQuery,
+    },
+    [sortBy]
+  );
+
+  const onReportSelected = useCallback(
+    (report: ReportRowFragment | undefined) => {
+      if (report === undefined) {
+        return;
+      }
+      if (report.argumentSchema) {
+        setReportWithArgs(report);
+      } else {
+        printReport(report, undefined);
+      }
+    },
+    []
+  );
+
+  const { print, isPrinting } = useReport.utils.print();
+
+  const printReport = (
+    report: ReportRowFragment,
+    args: JsonData | undefined
+  ) => {
+    print({ reportId: report.id, dataId: '', args });
+  };
+
+  return (
+    <>
+      <Toolbar filter={filter} />
+      <DataTable
+        id="item-list"
+        pagination={{ ...pagination, total: data?.totalCount }}
+        onChangePage={updatePaginationQuery}
+        columns={columns}
+        data={data?.nodes}
+        isError={isError}
+        isLoading={isLoading || isPrinting}
```

Comment:
Future work: does this need a better message when loading? i.e. that it is printing?

Reply:
yes! this was a clumsy interim solution. I don't like that it causes the list to effectively reload. 
I hadn't thought of a print button per line, but I like the idea. I'll see if we can do that, in which case a loading button would allow the combination of the two concerns to be solved together.

Reply:
I'm not entirely sold on the look of it.. but it is clearer!

<img width="1281" alt="Screenshot 2023-02-28 at 2 12 55 PM" src="https://user-images.githubusercontent.com/9192912/221727420-6a14aadc-55aa-438b-9ef4-e1354f1dd61c.png">

Reply:
nice! thanks!